validation/validate/search.go

```diff
@@ -0,0 +1,34 @@
+package validate
+
+import (
+	"unicode"
+
+	"github.com/target/goalert/validation"
+)
+
+// Search will validate a search body. It ensures that the field
+// consists of valid unicode code-points, and does not exceed max of 255 characters.
+// If body is empty, the input is considered valid.
+func Search(fname, body string) error {
+	if body == "" {
+		return nil
+	}
+
+	r := []rune(body)
+	l := len(r)
+
+	if l == 0 {
```

Comment:
This is redundant, we do an empty check at the top, and the `range` below would be a no-op anyways. We can move the `len()` bit to the `if` statement below that way too (since we only need to check it once).